CMake FMU build missing include for external C library

### Description

FMU docker export fails when Modelica library is used in the model.

### Steps to Reproduce
Export the following code to FMU v2.0 co-simulation with `x86_64-linux-gnu docker run --pull=never multiarch/crossbuild` selected.
```modelica
model Test
import print = Modelica.Utilities.Streams.print;
equation
when time > 1 then
print("Hello");
end when;
end Test;
```

### Expected Behavior

FMU is exported.

### Version and OS

 - OpenModelica Version: v.1.19.2
 - OS: Windows 10 with docker installed.
 - Versions of used Modelica libraries if applicable: 4.0+maint.om

OpenModelica generates the following error:
```
[1] 13:33:38 Translation Notification
OS is Windows and docker platform detected. Using CMake to build FMU.

[2] 13:33:38 Translation Error
Error building simulator. Build log: docker run  --rm -w /fmu -v d1fc96f56ac313240c085edcec122933ed9baa73a32e9d57929d8d7d15a5f2e0:/fmu -e CROSS_TRIPLE=x86_64-linux-gnu --pull=never multiarch/crossbuild sh -c "cd "/fmu/Test.fmutmp/sources/" && mkdir build_cmake_x86_64-linux-gnu && cd build_cmake_x86_64-linux-gnu && cmake -DFMI_INTERFACE_HEADER_FILES_DIRECTORY=/fmu/fmiInclude -DCMAKE_BUILD_TYPE=Release .. && cmake --build . && make install && cd .. && rm -rf build_cmake_x86_64-linux-gnu" failed:
docker volume create 
d1fc96f56ac313240c085edcec122933ed9baa73a32e9d57929d8d7d15a5f2e0
docker run --cidfile Test.fmutmp.cidfile -v d1fc96f56ac313240c085edcec122933ed9baa73a32e9d57929d8d7d15a5f2e0:/data busybox true 
docker cp Test.fmutmp 22f2dadd38821277ddd4bf1c405599a3d936ef6dc005a9af9e12af50cfa6c673:/data 
docker cp "C:/Program Files/OpenModelica1.19.2-64bit/include/omc/c/fmi" 22f2dadd38821277ddd4bf1c405599a3d936ef6dc005a9af9e12af50cfa6c673:/data/fmiInclude 
docker run  --rm -w /fmu -v d1fc96f56ac313240c085edcec122933ed9baa73a32e9d57929d8d7d15a5f2e0:/fmu -e CROSS_TRIPLE=x86_64-linux-gnu --pull=never multiarch/crossbuild sh -c "cd "/fmu/Test.fmutmp/sources/" && mkdir build_cmake_x86_64-linux-gnu && cd build_cmake_x86_64-linux-gnu && cmake -DFMI_INTERFACE_HEADER_FILES_DIRECTORY=/fmu/fmiInclude -DCMAKE_BUILD_TYPE=Release .. && cmake --build . && make install && cd .. && rm -rf build_cmake_x86_64-linux-gnu" 
-- The C compiler identification is GNU 6.3.0
-- The CXX compiler identification is GNU 6.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Building for FMI platform linux64
-- Configuring done
-- Generating done
-- Build files have been written to: /fmu/Test.fmutmp/sources/build_cmake_x86_64-linux-gnu
Scanning dependencies of target Test
[  1%] Building C object CMakeFiles/Test.dir/gc/memory_pool.c.o
[  3%] Building C object CMakeFiles/Test.dir/gc/omc_gc.c.o
[  4%] Building C object CMakeFiles/Test.dir/math-support/pivot.c.o
[  6%] Building C object CMakeFiles/Test.dir/meta/meta_modelica_catch.c.o
[  7%] Building C object CMakeFiles/Test.dir/simulation/omc_simulation_util.c.o
[  9%] Building C object CMakeFiles/Test.dir/simulation/options.c.o
[ 10%] Building C object CMakeFiles/Test.dir/simulation/simulation_info_json.c.o
[ 12%] Building C object CMakeFiles/Test.dir/simulation/simulation_omc_assert.c.o
/fmu/Test.fmutmp/sources/simulation/simulation_omc_assert.c: In function 'omc_assert_simulation':
/fmu/Test.fmutmp/sources/simulation/simulation_omc_assert.c:115:1: warning: 'noreturn' function does return
 } 
 ^
/fmu/Test.fmutmp/sources/simulation/simulation_omc_assert.c: In function 'omc_assert_simulation_withEquationIndexes':
/fmu/Test.fmutmp/sources/simulation/simulation_omc_assert.c:123:1: warning: 'noreturn' function does return
 } 
 ^
[ 13%] Building C object CMakeFiles/Test.dir/simulation/solver/delay.c.o
[ 15%] Building C object CMakeFiles/Test.dir/simulation/solver/fmi_events.c.o
[ 16%] Building C object CMakeFiles/Test.dir/simulation/solver/initialization/initialization.c.o
[ 18%] Building C object CMakeFiles/Test.dir/simulation/solver/model_help.c.o
[ 19%] Building C object CMakeFiles/Test.dir/simulation/solver/omc_math.c.o
[ 21%] Building C object CMakeFiles/Test.dir/simulation/solver/spatialDistribution.c.o
[ 22%] Building C object CMakeFiles/Test.dir/simulation/solver/stateset.c.o
[ 24%] Building C object CMakeFiles/Test.dir/simulation/solver/synchronous.c.o
[ 25%] Building C object CMakeFiles/Test.dir/util/ModelicaUtilities.c.o
/fmu/Test.fmutmp/sources/util/ModelicaUtilities.c: In function 'ModelicaError':
/fmu/Test.fmutmp/sources/util/ModelicaUtilities.c:83:1: warning: 'noreturn' function does return
 } 
 ^
/fmu/Test.fmutmp/sources/util/ModelicaUtilities.c: In function 'ModelicaVFormatError':
/fmu/Test.fmutmp/sources/util/ModelicaUtilities.c:87:1: warning: 'noreturn' function does return
 } 
 ^
/fmu/Test.fmutmp/sources/util/ModelicaUtilities.c: In function 'ModelicaFormatError':
/fmu/Test.fmutmp/sources/util/ModelicaUtilities.c:94:1: warning: 'noreturn' function does return
 } 
 ^
[ 27%] Building C object CMakeFiles/Test.dir/util/base_array.c.o
[ 28%] Building C object CMakeFiles/Test.dir/util/boolean_array.c.o
[ 30%] Building C object CMakeFiles/Test.dir/util/division.c.o
[ 31%] Building C object CMakeFiles/Test.dir/util/doubleEndedList.c.o
[ 33%] Building C object CMakeFiles/Test.dir/util/generic_array.c.o
[ 34%] Building C object CMakeFiles/Test.dir/util/index_spec.c.o
[ 36%] Building C object CMakeFiles/Test.dir/util/integer_array.c.o
[ 37%] Building C object CMakeFiles/Test.dir/util/jacobian_util.c.o
[ 39%] Building C object CMakeFiles/Test.dir/util/list.c.o
[ 40%] Building C object CMakeFiles/Test.dir/util/modelica_string.c.o
[ 42%] Building C object CMakeFiles/Test.dir/util/modelica_string_lit.c.o
[ 43%] Building C object CMakeFiles/Test.dir/util/omc_error.c.o
[ 45%] Building C object CMakeFiles/Test.dir/util/omc_file.c.o
[ 46%] Building C object CMakeFiles/Test.dir/util/omc_init.c.o
[ 48%] Building C object CMakeFiles/Test.dir/util/omc_mmap.c.o
[ 50%] Building C object CMakeFiles/Test.dir/util/omc_msvc.c.o
[ 51%] Building C object CMakeFiles/Test.dir/util/omc_numbers.c.o
[ 53%] Building C object CMakeFiles/Test.dir/util/parallel_helper.c.o
[ 54%] Building C object CMakeFiles/Test.dir/util/rational.c.o
[ 56%] Building C object CMakeFiles/Test.dir/util/real_array.c.o
[ 57%] Building C object CMakeFiles/Test.dir/util/ringbuffer.c.o
[ 59%] Building C object CMakeFiles/Test.dir/util/simulation_options.c.o
[ 60%] Building C object CMakeFiles/Test.dir/util/string_array.c.o
[ 62%] Building C object CMakeFiles/Test.dir/util/utility.c.o
[ 63%] Building C object CMakeFiles/Test.dir/util/varinfo.c.o
[ 65%] Building C object CMakeFiles/Test.dir/Test.c.o
[ 66%] Building C object CMakeFiles/Test.dir/Test_01exo.c.o
[ 68%] Building C object CMakeFiles/Test.dir/Test_02nls.c.o
[ 69%] Building C object CMakeFiles/Test.dir/Test_03lsy.c.o
[ 71%] Building C object CMakeFiles/Test.dir/Test_04set.c.o
[ 72%] Building C object CMakeFiles/Test.dir/Test_05evt.c.o
[ 74%] Building C object CMakeFiles/Test.dir/Test_06inz.c.o
[ 75%] Building C object CMakeFiles/Test.dir/Test_07dly.c.o
[ 77%] Building C object CMakeFiles/Test.dir/Test_08bnd.c.o
[ 78%] Building C object CMakeFiles/Test.dir/Test_09alg.c.o
[ 80%] Building C object CMakeFiles/Test.dir/Test_10asr.c.o
[ 81%] Building C object CMakeFiles/Test.dir/Test_11mix.c.o
[ 83%] Building C object CMakeFiles/Test.dir/Test_12jac.c.o
[ 84%] Building C object CMakeFiles/Test.dir/Test_13opt.c.o
[ 86%] Building C object CMakeFiles/Test.dir/Test_14lnz.c.o
[ 87%] Building C object CMakeFiles/Test.dir/Test_15syn.c.o
[ 89%] Building C object CMakeFiles/Test.dir/Test_16dae.c.o
[ 90%] Building C object CMakeFiles/Test.dir/Test_17inl.c.o
[ 92%] Building C object CMakeFiles/Test.dir/Test_18spd.c.o
[ 93%] Building C object CMakeFiles/Test.dir/Test_FMU.c.o
In file included from /fmu/Test.fmutmp/sources/Test_FMU.c:59:0:
/fmu/Test.fmutmp/sources/fmi-export/fmu2_model_interface.c.inc: In function 'omc_assert_fmi_common':
/fmu/Test.fmutmp/sources/fmi-export/fmu2_model_interface.c.inc:213:16: warning: passing argument 1 of 'GC_vasprintf' from incompatible pointer type [-Wincompatible-pointer-types]
   GC_vasprintf(&str, msg, args); 
                ^
In file included from /fmu/Test.fmutmp/sources/openmodelica_func.h:52:0,
                 from /fmu/Test.fmutmp/sources/Test_FMU.c:10:
/fmu/Test.fmutmp/sources/util/modelica_string.h:71:5: note: expected 'const char **' but argument is of type 'char **'
 int GC_vasprintf(const char **strp, const char *fmt, va_list ap); 
     ^~~~~~~~~~~~
[ 95%] Building C object CMakeFiles/Test.dir/Test_functions.c.o
In file included from /fmu/Test.fmutmp/sources/Test_functions.c:7:0:
/fmu/Test.fmutmp/sources/Test_includes.h:4:30: fatal error: ModelicaInternal.h: No such file or directory
 #include "ModelicaInternal.h" 
                              ^
compilation terminated.
make[2]: *** [CMakeFiles/Test.dir/Test_functions.c.o] Error 1
CMakeFiles/Test.dir/build.make:1550: recipe for target 'CMakeFiles/Test.dir/Test_functions.c.o' failed
make[1]: *** [CMakeFiles/Test.dir/all] Error 2
make: *** [all] Error 2
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/Test.dir/all' failed
Makefile:127: recipe for target 'all' failed
```

--- a/OMCompiler/Compiler/SimCode/SimCodeMain.mo
+++ b/OMCompiler/Compiler/SimCode/SimCodeMain.mo
@@ -916,8 +916,9 @@ algorithm
           cmakelistsStr := System.stringReplace(cmakelistsStr, "@WITH_SUNDIALS@", "");
         end if;
 
-        // Check for external libraries
+        // Add external libraries and includes
         cmakelistsStr := System.stringReplace(cmakelistsStr, "@FMU_ADDITIONAL_LIBS@", SimCodeUtil.getCmakeLinkLibrariesCode(simCode.makefileParams.libs));
+        cmakelistsStr := System.stringReplace(cmakelistsStr, "@FMU_ADDITIONAL_INCLUDES@", SimCodeUtil.make2CMakeInclude(simCode.makefileParams.includes));
 
         System.writeFile(fmu_tmp_sources_dir + "CMakeLists.txt", cmakelistsStr);
 

--- a/OMCompiler/Compiler/SimCode/SimCodeUtil.mo
+++ b/OMCompiler/Compiler/SimCode/SimCodeUtil.mo
@@ -15469,5 +15469,16 @@ algorithm
   end for;
 end getCmakeLinkLibrariesCode;
 
+public function make2CMakeInclude
+  "Convert makefile include directories to CMake include directories"
+  input list<String> includes;
+  output String cmakecode = "";
+algorithm
+  for include in includes loop
+    cmakecode := cmakecode + "\n                                               " +
+                 "\"" + System.trim(include, "\"-I") + "\"";
+  end for;
+end make2CMakeInclude;
+
 annotation(__OpenModelica_Interface="backend");
 end SimCodeUtil;

--- a/OMCompiler/SimulationRuntime/fmi/export/buildproject/CMakeLists.txt.in
+++ b/OMCompiler/SimulationRuntime/fmi/export/buildproject/CMakeLists.txt.in
@@ -54,7 +54,7 @@ target_link_libraries(${FMU_NAME} PRIVATE Threads::Threads)
 
 target_include_directories(${FMU_NAME} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 target_include_directories(${FMU_NAME} PRIVATE ${FMI_INTERFACE_HEADER_FILES_DIRECTORY}
-                                               sundials)
+                                               sundials@FMU_ADDITIONAL_INCLUDES@)
 
 target_compile_definitions(${FMU_NAME} PRIVATE OMC_MINIMAL_RUNTIME=1;OMC_FMI_RUNTIME=1;CMINPACK_NO_DLL)
 

